contact_map/plot_utils.py

```diff
@@ -3,10 +3,126 @@
 try:  # try loop for testing
     import matplotlib
     import matplotlib.pyplot as plt
-    from matplotlib.colors import LinearSegmentedColormap
+    from matplotlib.colors import LinearSegmentedColormap, Normalize
 except ImportError:  # pragma: no cover
     pass
 
+def _get_cmap(cmap):
+    """Get the appropriate matplotlib colormap.
+
+    If the input is a string, it finds the associated matplotlib ColorMap
+    object. If input is already a ColorMap, it is passed through.
+    """
+    if isinstance(cmap, str):
+        cmap_f = plt.get_cmap(cmap)
+    else:
+        cmap_f = cmap
+
+    return cmap_f
+
+
+def _make_norm(rescale_range='auto'):
+    """Create the matplotlib Normalize object
+    """
+    if rescale_range is None:
+        norm = Normalize(vmin=0, vmax=1, clip=True)
+    elif rescale_range == 'auto':
+        norm = Normalize()
+    elif isinstance(rescale_range, Normalize):
+        norm = rescale_range
+    else:
+        vmin, vmax = rescale_range
+        norm = Normalize(vmin, vmax, clip=True)
+
+    return norm
+
+
+def rgb_colors(data, cmap, rescale_range='auto', bytes=False):
+    """RGB values representing colors after applying a colormap to data.
+
+    Parameters
+    ----------
+    data : array-like
+        Input data, should be 1-dimensional
+    cmap : str or matplotlib.colors.Colormap
+        Colormap to use
+    rescale_range : matplotlib.colors.Normalize or str or None or tuple
+        Defines the color normalization. If a ``Normalize`` object, that is
+        passed through. The string 'auto' is interpresent as
+        ``Normalize()``. No other strings are currently allowed. ``None`` is
+        interpreted as ``Normalize(0, 1)``, meaning that the data is already
+        mapped to the correct values. A tuple should be a 2-tuple of (vmin,
+        vmax).
+    bytes : bool
```

Comment:
please add slightly more documentation to this keyword (as it was relatively difficult to find in the matplotlib documentation as well)